lib/grandpa/message_handler.go

```diff
@@ -284,22 +284,21 @@ func (h *MessageHandler) verifyCatchUpResponseCompletability(prevote, precommit
 	return nil
 }
 
-func getEquivocatoryVoters(votes []AuthData) (map[ed25519.PublicKeyBytes]struct{}, error) {
+func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{} {
 	eqvVoters := make(map[ed25519.PublicKeyBytes]struct{})
-	voters := make(map[ed25519.PublicKeyBytes]int, len(votes))
+	voters := make(map[ed25519.PublicKeyBytes][64]byte, len(votes))
 
 	for _, v := range votes {
-		voters[v.AuthorityID]++
-		switch voters[v.AuthorityID] {
-		case 1:
-		case 2:
-			eqvVoters[v.AuthorityID] = struct{}{}
-		default:
-			return nil, fmt.Errorf("%w: authority id %x has %d votes",
-				errInvalidMultiplicity, v.AuthorityID, voters[v.AuthorityID])
+		signature, present := voters[v.AuthorityID]
+		if present {
+			if !bytes.Equal(signature[:], v.Signature[:]) {
```

Comment:
If there is a vote present for the authority, and the signature of that vote doesn't equal the votes signature then we add that voter to the equivocatory voters map.  The voter is equivocatory since there are votes for that voter that do not match.